MaterialDesignThemes.UITests/WPF/DatePickers/DatePickerTests.cs

```diff
@@ -64,5 +64,75 @@ public async Task OnDatePicker_WithClearButton_ClearsSelectedDate()
             
             recorder.Success();
         }
+
+        [Fact]
+        [Description("Issue 2737")]
+        public async Task OutlinedDatePicker_RespectsActiveAndInactiveBorderThickness_WhenAttachedPropertiesAreSet()
+        {
+            await using var recorder = new TestRecorder(App);
+
+            // Arrange
+            var expectedInactiveBorderThickness = new Thickness(4, 3, 2, 1);
+            var expectedActiveBorderThickness = new Thickness(1, 2, 3, 4);
+            var stackPanel = await LoadXaml<StackPanel>($@"
+<StackPanel>
+    <DatePicker Style=""{{StaticResource MaterialDesignOutlinedDatePicker}}""
+      materialDesign:DatePickerAssist.OutlinedBorderInactiveThickness=""{expectedInactiveBorderThickness}""
+      materialDesign:DatePickerAssist.OutlinedBorderActiveThickness=""{expectedActiveBorderThickness}"">
+      <DatePicker.SelectedDate>
+        <Binding RelativeSource=""{{RelativeSource Self}}"" Path=""Tag"" UpdateSourceTrigger=""PropertyChanged"">
+          <Binding.ValidationRules>
+            <local:FutureDateValidationRule ValidatesOnTargetUpdated=""True""/>
+          </Binding.ValidationRules>
+        </Binding>
+      </DatePicker.SelectedDate>
+    </DatePicker>
+    <Button x:Name=""Button"" Content=""Some Button"" Margin=""0,20,0,0"" />
+</StackPanel>", ("local", typeof(FutureDateValidationRule)));
+            var datePicker = await stackPanel.GetElement<DatePicker>("/DatePicker");
+            await datePicker.SetProperty(DatePicker.SelectedDateProperty, DateTime.Now.AddDays(1));
+            var datePickerTextBox = await datePicker.GetElement<TextBox>("PART_TextBox");
+            var button = await stackPanel.GetElement<Button>("Button");
+
+            // Act
+            await button.MoveCursorTo();
+            await Task.Delay(50);   // Wait for the visual change
+            var inactiveBorderThickness = await datePickerTextBox.GetProperty<Thickness>(Control.BorderThicknessProperty);
+            await datePickerTextBox.MoveCursorTo();
+            await Task.Delay(50);   // Wait for the visual change
+            var hoverBorderThickness = await datePickerTextBox.GetProperty<Thickness>(Control.BorderThicknessProperty);
+            await datePickerTextBox.LeftClick();
+            await Task.Delay(50);   // Wait for the visual change
+            var focusedBorderThickness = await datePickerTextBox.GetProperty<Thickness>(Control.BorderThicknessProperty);
+
+            // TODO: It would be cool if a validation error could be set via XAMLTest without the need for the Binding and ValidationRules elements in the XAML above.
```

Comment:
This is a good request. I will have to look into what that API could look like. I typically have done that by building out a custom control in the test project (as this then supports things like code behind, view models, etc). But I can certainly see a desire to not have to jump through all of those hoops.

Reply:
> This is a good request. I will have to look into what that API could look like. I typically have done that by building out a custom control in the test project (as this then supports things like code behind, view models, etc). But I can certainly see a desire to not have to jump through all of those hoops.

Cool! As mentioned in the PR, I already have a proof-of-concept PR in the XAMLTest library with one approach to adding this type of functionality. Looking forward to seeing how you would approach it.